javascript-sdk/src/sdk.ts

```diff
@@ -28,38 +27,45 @@ export default class RebuffSdk implements Rebuff {
   private strategies: Record<string, Strategy> | undefined;
   private defaultStrategy: string;
 
-  private openai: {
-    conn: OpenAIApi;
-    model: string;
-  };
-
+  /**
+   * @deprecated Use `RebuffSdk.init` instead.
+   */
   constructor(config: SdkConfig) {
+    // We're keeping this constructor for backwards compatibility. In the future, we can make it private and
+    // simplify this class quite a bit.
+
     this.sdkConfig = config;
-    this.openai = {
-      conn: getOpenAIInstance(config.openai.apikey),
-      model: config.openai.model || "gpt-3.5-turbo",
-    };
     this.defaultStrategy = "standard";
   }
 
+  public static async init(config: SdkConfig): Promise<RebuffSdk> {
+    const sdk = new RebuffSdk(config);
+    sdk.vectorStore = await initVectorStore(config);
+    sdk.strategies = await sdk.getStrategies();
+    return sdk;
+  }
+
   private async getStrategies(): Promise<Record<string, Strategy>> {
     if (this.strategies) {
       return this.strategies;
     }
+    const openai = {
```

Comment:
Not a huge deal to me, but it may be better to create the openai connection in the init so that we don't have to create it every time we detect an injection or call a function that needs it.

Reply:
We'll only create the openai connection once per lifetime of `RebuffSdk`. The second time that `detectInjection` is called, the `if (this.strategies) {` condition will be met and it will return early.

Reply:
Makes sense, I missed that. Thanks